### PR TITLE
WIP: CppExport

### PR DESCRIPTION
**Pull request by patrick-luethi**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/336%23discussion_r54101128%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/336%23discussion_r54101384%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/336%23issuecomment-188817140%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/336%23issuecomment-188817140%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222016-02-25T14%3A49%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6049349%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/patrick-luethi%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b46725b415b4bf01991a46ce89de8df73de20a8d%20CppExport/src/exporter/CppExporter.cpp%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/336%23discussion_r54101384%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20a%20put%20a%20comment%20here%20on%20top%20to%20explain%20briefly%20what%27s%20going%20on.%20%28why%20is%20the%20code%20below%20necessary%29%22%2C%20%22created_at%22%3A%20%222016-02-25T14%3A26%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CppExport/src/exporter/CppExporter.cpp%3AL206-219%22%7D%2C%20%22Pull%20b46725b415b4bf01991a46ce89de8df73de20a8d%20CppExport/src/CodeComposite.h%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/336%23discussion_r54101128%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60additionalDependencies%60%22%2C%20%22created_at%22%3A%20%222016-02-25T14%3A25%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CppExport/src/CodeComposite.h%3AL55-70%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull b46725b415b4bf01991a46ce89de8df73de20a8d CppExport/src/CodeComposite.h 7'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/336#discussion_r54101128'>File: CppExport/src/CodeComposite.h:L55-70</a></b>
- <a href='https://github.com/dimitar-asenov'></a> `additionalDependencies`
- [ ] <a href='#crh-comment-Pull b46725b415b4bf01991a46ce89de8df73de20a8d CppExport/src/exporter/CppExporter.cpp 6'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/336#discussion_r54101384'>File: CppExport/src/exporter/CppExporter.cpp:L206-219</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Please a put a comment here on top to explain briefly what's going on. (why is the code below necessary)
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/336#issuecomment-188817140'>General Comment</a></b>
- <a href='https://github.com/patrick-luethi'></a> done

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/336?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/336?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/336'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/336'>Refresh</a>.<a href='#crh-update-none'></a>

File: CppExport/src/CodeComposite.h [hunk 1]
```diff
@@ -55,9 +55,15 @@ class CPPEXPORT_API CodeComposite
 
 		bool isTemplateImplementationSeparateFile();
 
+		bool isXMacroData();
+		bool isXMacroInstantiation();
+
+		QList<CodeUnit*>& additionalUnitsThisDependsOn();
```
Comment (dimitar-asenov) [inline CppExport/src/CodeComposite.h]: `additionalDependencies`

File: CppExport/src/exporter/CppExporter.cpp [hunk 2]
```diff
@@ -206,7 +206,13 @@ QList<CodeComposite*> CppExporter::mergeUnits(QList<CodeUnit*>& units, QHash<QSt
 		}
 	}
 
-	return nameToCompositeMap.values();
+	QList<CodeComposite*> result = nameToCompositeMap.values();
+	for (auto potentialMacroData : result)
```
Comment (dimitar-asenov) [inline CppExport/src/exporter/CppExporter.cpp:212]: Please a put a comment here on top to explain briefly what's going on. (why is the code below necessary)

**Comment by patrick-luethi** [comment]
done

**Pull request merged**